### PR TITLE
Us012 add new role to existing job family and band

### PR DESCRIPTION
**Pull request by gibpaul**

### CODE REVIEW

**Comment by gibpaul** [comment]
@ccurran01

**Review by ConnMeehan — APPROVED** [review]

**Review by philipwilson5 — APPROVED** [review]

File: src/main/java/org/kainos/ea/cli/BandRequest.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.kainos.ea.cli;
+
+public class BandRequest {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/cli/BandRequest.java]: looks like this isn't needed?

File: src/main/java/org/kainos/ea/service/CapabilityService.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public List<Capability> getAllCapabilities() throws FailedToGetCapabilitiesException, DatabaseConnectionException {
+        try {
+            return capabilityDao.getAllCapabilities(databaseConnector.getConnection());
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+            throw new FailedToGetCapabilitiesException();
+        } catch (DatabaseConnectionException e) {
+            System.err.println(e.getMessage());
+            throw new FailedToGetCapabilitiesException();
+        }
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/service/CapabilityService.java]: can you merge these 2 catch blocks into 1 catch block

File: src/main/java/org/kainos/ea/validator/BandValidator.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.kainos.ea.validator;
+
+import org.kainos.ea.model.Band;
+
+public class BandValidator {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/validator/BandValidator.java]: not sure this is needed?

File: src/main/java/org/kainos/ea/validator/JobRoleValidator.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package org.kainos.ea.validator;
+
+import org.kainos.ea.cli.JobRoleRequest;
+
+public class JobRoleValidator {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/validator/JobRoleValidator.java:12]: can you also check if the capability ID and band ID exist in the database

File: src/test/java/org/kainos/ea/service/JobRoleServiceTest.java [hunk 5]
```diff
@@ -59,6 +63,68 @@ void getJobRolesShouldThrowDatabaseConnectionExceptionWhenDaoThrowsDatabaseConne
 
     }
 
+    @Test
```
Comment (shaunganley) [inline src/test/java/org/kainos/ea/service/JobRoleServiceTest.java:65]: you're not mocking out your validator anywhere in here. need a unit test for what happens when validation fails

**Review by shaunganley — CHANGES_REQUESTED** [review]
great start, some small comments and needs integration tests

**Review by shaunganley — APPROVED** [review]

**Pull request merged**